unlock-app/src/components/interface/checkout/Lock.tsx

```diff
@@ -55,16 +58,18 @@ export const Lock = ({
 
   const canAfford = userCanAffordKey(lock, balances)
 
+  const keyForThisLock = activeKeys.find(key => key.lock === lock.address)
```

Comment:
I wonder if this could not happen one level above? But I hguess this creates trouble to detect that a lock should be "disabled"?

Reply:
Yup, that's exactly the thought process I had. I was going to just pass a boolean, but for disabling other locks it was easiest to just pass the list down.